src/include/sof/audio/format.h

```diff
@@ -155,7 +155,7 @@ static inline int32_t sign_extend_s24(int32_t x)
 
 static inline uint32_t get_sample_bytes(enum sof_ipc_frame fmt)
 {
-	return fmt == SOF_IPC_FRAME_S16_LE ? 2 : 4;
+	return fmt == SOF_IPC_FRAME_S16_LE ? 2 :  fmt == SOF_IPC_FRAME_S24_3LE ? 3 : 4;
```

Comment:
this looks so bad, use a switch to make it easier to read

Reply:
sure